engine/src/main/java/com/google/android/fhir/sync/SyncJobImpl.kt

```diff
@@ -60,16 +61,28 @@ class SyncJobImpl(private val context: Context) : SyncJob {
           periodicSyncConfiguration.repeat.timeUnit
         )
         .setConstraints(periodicSyncConfiguration.syncConstraints)
-        .build()
 
+    if (periodicSyncConfiguration.retryConfiguration?.backoffCriteria != null) {
+      periodicWorkRequest
+        .setBackoffCriteria(
+          periodicSyncConfiguration.retryConfiguration.backoffCriteria.backoffPolicy,
+          periodicSyncConfiguration.retryConfiguration.backoffCriteria.backoffDelay,
+          periodicSyncConfiguration.retryConfiguration.backoffCriteria.timeUnit
+        )
+        .setInputData(
+          Data.Builder()
+            .putInt(MAX_RETRIES_ALLOWED, periodicSyncConfiguration.retryConfiguration.maxRetries)
+            .build()
+        )
+    }
```

Comment:
`RetryConfiguration.backoffCriteria` is `non-null` type.
```suggestion
    periodicSyncConfiguration.retryConfiguration?.let {
      periodicWorkRequest
        .setBackoffCriteria(
       it.backoffCriteria.backoffPolicy,
        it.backoffCriteria.backoffDelay,
        it.backoffCriteria.timeUnit
        )
        .setInputData(
          Data.Builder()
            .putInt(MAX_RETRIES_ALLOWED, it.maxRetries)
            .build()
        )
    }
```